cluster.go

```diff
@@ -1076,6 +1078,53 @@ func (c *Cluster) unpinClusterDag(metaPin api.Pin) error {
 	return nil
 }
 
+// PinPath accepts a path string resolves it into a cid and makes cluster pin it
+func (c *Cluster) PinPath(path string, p api.Pin) (cid.Cid, error) {
+	ci, err := c.resolve(path)
+	if err != nil {
+		return ci, err
+	}
+
+	p.Cid = ci
+	return ci, c.Pin(p)
+}
+
+// UnpinPath accepts a path string resolves it into a cit and makes the cluster unpin it
+func (c *Cluster) UnpinPath(path string) (cid.Cid, error) {
+	ci, err := c.resolve(path)
+	if err != nil {
+		return ci, err
+	}
+	return ci, c.Unpin(ci)
+}
+
+// resolve resolves given string path into a cid
+func (c *Cluster) resolve(path string) (cid.Cid, error) {
+	var ci cid.Cid
+	validPath, err := gopath.ParsePath(path)
+	if err != nil {
+		logger.Error("could not parse path")
+		return cid.Undef, err
+	}
+
+	if !strings.HasPrefix(path, "/ipns") && validPath.IsJustAKey() {
+		ci, _, err = gopath.SplitAbsPath(validPath)
+		if err != nil {
```

Comment:
we could `return ci, err` here and avoid all the indenting below (and having to declare `ci` at the top..